src/Sentry/Platforms/Cocoa/RuntimeMarshalManagedExceptionIntegration.cs

```diff
@@ -0,0 +1,45 @@
+using ObjCRuntime;
+using Sentry.Extensibility;
+using Sentry.Integrations;
+
+namespace Sentry.Cocoa;
+
+/// <summary>
+/// When AOT Compiling iOS applications, the AppDomain UnhandledExceptionHandler doesn't fire. So instead we intercept
+/// the Runtime.RuntimeMarshalManagedException event.
+/// </summary>
+internal class RuntimeMarshalManagedExceptionIntegration : ISdkIntegration
+{
+    private readonly IRuntime _runtime;
+    private IHub? _hub;
+    private SentryOptions? _options;
+
+    internal RuntimeMarshalManagedExceptionIntegration(IRuntime? runtime = null)
+        => _runtime = runtime ?? RuntimeAdapter.Instance;
+
+    public void Register(IHub hub, SentryOptions options)
+    {
+        _hub = hub;
+        _options = options;
+        _runtime.MarshalManagedException += Handle;
+    }
+
+    // Internal for testability
+    [SecurityCritical]
+    internal void Handle(object sender, MarshalManagedExceptionEventArgs e)
+    {
+        _options?.LogDebug("Runtime Marshal Managed Exception");
+
+        if (e.Exception is { } ex)
+        {
+            ex.SetSentryMechanism(
+                "Runtime.MarshalManagedException",
+                "This exception was caught by the .NET Runtime Marshal Managed Exception global error handler. " +
+                "The application may have crashed as a result of this exception.",
```

Comment:
It's not certain that it did crash?

Reply:
I'm not sure about that no. We're using this event instead of `AppDomain.Current.UnhandledException` because `Runtime.MarshalManagedException` fires even in AOT compiled applications. 

When testing, it doesn't fire for exceptions that are caught by managed code (in a `try..catch` block) so I I think it ordinarily would crash the application. I don't know enough about the interface between .NET and ObjectiveC to be sure of this though.

Reply:
But regardless of that bool, if that fires (on your tests) the app crashes? If so, I'd indicate that's the case. But keeping it vague in here is fine, I was just wondering how this works